src/js/common/components/RequiredLoginView.jsx

```diff
@@ -5,22 +5,41 @@ import { intersection } from 'lodash';
 
 import SystemDownView from './SystemDownView';
 import LoadingIndicator from '../../common/components/LoadingIndicator';
-import Main from '../../login/containers/Main';
 
 const healthTools = ['health-records', 'rx', 'messaging'];
 const nextQuery = { next: window.location.pathname };
 const signInUrl = appendQuery('/', nextQuery);
+const verifyUrl = appendQuery('/verify', nextQuery);
 
 class RequiredLoginView extends React.Component {
   componentDidUpdate() {
     const isLoading = this.props.user.profile.loading;
-    if (!isLoading && this.shouldSignIn()) {
-      window.location.replace(signInUrl);
+
+    if (!isLoading) {
+      if (this.shouldSignIn()) {
+        window.location.replace(signInUrl);
+      } else if (this.shouldVerify()) {
+        window.location.replace(verifyUrl);
```

Comment:
Didn't think there was a good reason to render the verify app inline but still have redirect logic in the verify component. So just being consistent by always having it redirect, similar how we would if we needed the user to log in.